include/ccf/endpoint_registry.h

```diff
@@ -130,8 +130,8 @@ namespace ccf::endpoints
     ccf::pal::Mutex metrics_lock;
     std::map<std::string, std::map<std::string, Metrics>> metrics;
 
-    EndpointRegistry::Metrics& get_metrics_for_endpoint(
-      const EndpointDefinitionPtr& e);
+    EndpointRegistry::Metrics& get_metrics_for_request(
```

Comment:
Is this an API change?

Reply:
Ach, good spot! Hadn't realised this was public. This change is necessary so that we can increment the number of calls once, before endpoint lookup. I'll add a note to the changelog

Reply:
I've mentioned this in the changelog, but the better fix would be to remove this (and more importantly the `increment_metrics_*` calls) from the public API. Tracked in #4207.